### PR TITLE
Implement enhanced logging and monitoring features

### LINKED ISSUE #48: Implementer logging

### PR DESCRIPTION
**Pull request by AeselCSS**
## Description

Improved logging throughout the application, including handlers, middleware, and utility functions. Added configuration for Prometheus metrics and monitoring services. Enhanced error handling and logging for various operations, including user authentication and environment configuration loading. Updated Docker Compose for log management and monitoring.

### Issues # (issue-id)

Fixes #48

### Additional comments

none

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Code quality improvements (linting, refactoring, etc.)
- [ ] Other (please specify):

## Checklist

- [x] My code follows the style guidelines of this project (ESLint, GoLint, etc.).
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have run relevant linters (ESLint, GoLint, Qodana, etc.) and ensured that no new issues were introduced.

### CODE REVIEW

File: backend/internal/api/handlers/login_handler.go
Comment (emilvn) [inline backend/internal/api/handlers/login_handler.go:1]: Kunne man ikke genbruge "sanitizedUsername" i stedet for at gøre det samme 3 gange i samme funktion?

File: backend/internal/utils/logger.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+}
+
+// ObfuscateSensitiveFields obfuscates sensitive fields in log messages
+func ObfuscateSensitiveFields(fields logrus.Fields) logrus.Fields {
+	obfuscatedFields := make(logrus.Fields)
+	for key, value := range fields {
+		if key == "jwt" {
+			strValue, ok := value.(string)
+			if ok && len(strValue) > 10 {
+				obfuscatedFields[key] = strValue[:5] + "*****" + strValue[len(strValue)-5:]
```
Comment (emilvn) [inline backend/internal/utils/logger.go:48]: Vil vi gerne vise de 5 karakterer af jwt? Kan godt se det giver mening i forhold til debugging at man kan se den ikke har et eller andet mærkeligt, men er i tvivl om sikkerheden med at vise noget af den i logs.

File: backend/internal/utils/logger.go
Comment (emilvn) [inline backend/internal/utils/logger.go:1]: Er de warnings som gh advanced security giver for denne fil addresseret?

**Review by emilvn — CHANGES_REQUESTED** [review]
Ser umiddelbart nice ud. Fedt med solid logging til bedre debugging. 
Inden du merger må du gerne lige gå det hele igennem og fjerne/ændre i unødige kommentarer, der er en del.

File: backend/internal/utils/logger.go (hunk 1 above)
Comment (AeselCSS) [reply backend/internal/utils/logger.go:68]: skal vi bare undlade den fuldstændig? - det vil gøre mit liv noget lettere tbh

Comment (emilvn) [reply backend/internal/utils/logger.go:68]: Jo bare fjern den helt hvis det er nemmere

Comment (AeselCSS) [reply backend/internal/utils/logger.go:68]: Det er jo roden til at GitHub Advanced security botten fik noget at lave, så det er sgu nok det sikreste bare at fjerne det.

File: backend/internal/utils/logger.go
Comment (AeselCSS) [reply backend/internal/utils/logger.go:1]: Ja, nu er de.

File: backend/internal/api/handlers/login_handler.go
Comment (AeselCSS) [reply backend/internal/api/handlers/login_handler.go:1]: har lavet en sanitize util og bruger den flere steder foruden her.

**Comment by emilvn** [comment]
@AeselCSS Ignorerer vi de sidste par codeQL warnings?

**Comment by AeselCSS** [comment]
> @AeselCSS Ignorerer vi de sidste par codeQL warnings?

2/3 er false positives og den sidste har jeg lige fixet... har overset en auth header der blev sendt til loggen, men som nu er sat til nil

**Review by emilvn — APPROVED** [review]

**Pull request merged**